### PR TITLE
feat: support non drs environments

### LINKED ISSUE #458: Add support for ESXi Hosts in non-cluster vSphere environments
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Description

In some environments, there is no cluster between hosts, so I suggest supporting Hosts, instead of passing a cluster

### Use Case(s)

for example, if you have a home lab with 1 vcenter and 1 esxi.

### Potential Configuration

I have tested the following with `vCenter 8.0` and `vSphere ESXi 8.0` and `Packer 1.8.5`
```hcl
source "vsphere-iso" "linux-rocky" {
...
  cluster   = var.cluster_enabled ? var.vsphere_cluster : null
  host       = var.cluster_enabled ? null : var.vsphere_host
...
}
```

### References

https://developer.hashicorp.com/packer/plugins/builders/vsphere/vsphere-iso#working-with-clusters-and-hosts

### PR DESCRIPTION
**Pull request by tintinhamans**
# 

## Summary of Pull Request

Adds the option to configure the following target for builds: 
  - a vSphere cluster with Distributed Resource Scheduling enabled. (Default)
  - an ESXi host in vSphere cluster with Distributed Resource Scheduling disabled. (Override)
  - an ESXi host not in a vSphere cluster. (Override)

## Type of Pull Request

- [ ] This is a bugfix. `type/bug`
- [x] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.

## Related to Existing Issues

Resolves #458 

## Test and Documentation Coverage

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

## Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

**Comment by tintinhamans** [comment]
One thing that might also need to be adjusted is the Terraform part that creates the vSphere role, since you are not able to run that against a standalone ESXi host (without vCenter server) as the following permissions don't exist in that situation:

```hcl
  "ContentLibrary.AddLibraryItem",
  "ContentLibrary.UpdateLibraryItem",
  "Cryptographer.Access",
  "Cryptographer.Encrypt",
```

But I left that out of this PR since I felt it was out of scope.

**Review by tenthirtyam — DISMISSED** [review]
An update to `set-envvars.sh` is also required.

**Comment by tintinhamans** [comment]
> An update to `set-envvars.sh` is also required.

Updated as requested 😊

File: builds/linux/almalinux/8/variables.pkr.hcl [hunk 1]
```diff
@@ -42,6 +42,11 @@ variable "vsphere_cluster" {
   description = "The name of the target vSphere cluster. (e.g. 'sfo-w01-cl01')"
 }
 
+variable "vsphere_host" {
+  type        = string
+  description = "The name of the target vSphere host. (e.g. 'sfo-w01-esx01')"
```
Comment (tenthirtyam) [inline builds/linux/almalinux/8/variables.pkr.hcl:47]: “ESXi host” is the correct product term.

Comment (tintinhamans) [reply builds/linux/almalinux/8/variables.pkr.hcl]: Modified the descriptions, think that the variable name should stay `vsphere_host` for consistency with the other vSphere related variables.

**Comment by tintinhamans** [comment]
I just realized that there are two 'versions' of standalone host and did some additional testing:

1. Standalone host (without vCenter server):

`vsphere_datacenter` can either be set to `ha-datacenter` or it can also be left blank.

2. Standalone host (with vCenter server):

`vsphere_datacenter` should be set, `vsphere_folder` can be set.

I also set the defaults for these variables to empty strings since then people can then simply comment or remove the variables from vsphere.pkrvars.hcl altogether instead of having to set them empty.

File: README.md [hunk 2]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
 vsphere_folder               = "sfo-w01-fd-templates"
 ```
 
+> **Note**
+>
+> When targeting standalone hosts or clusters without DRS you must set the `vsphere_host` variable.
```
Comment (tenthirtyam) [inline README.md:774]:
```suggestion
> When targeting standalone ESXi hosts or vSphere clusters with vSphere DRS disabled, you must set the `vsphere_host` variable.
```

File: README.md [hunk 3]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+>
+> **Example** (Clusters without DRS):
```
Comment (tenthirtyam) [inline README.md:776]:
```suggestion
> **Example** (vSphere Clusters with vSphere DRS Disabled):
```

File: README.md [hunk 4]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> ```hcl
+> ...
+> vsphere_datacenter           = "sfo-w01-dc01"
+> vsphere_cluster              = "sfo-w01-cl01"
+> vsphere_host                 = "sfo-w01-esx01"
+> vsphere_folder               = "sfo-w01-fd-templates"
+> ...
+> ```
+>
+> **Example** (vCenter Managed Standalone Hosts):
```
Comment (tenthirtyam) [inline README.md:787]:
```suggestion
> **Example** (Standalone ESXi Hosts Managed by vCenter Server):
```

File: README.md [hunk 5]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+>
+> For vCenter managed standalone hosts, comment or remove `vsphere_cluster`.
```
Comment (tenthirtyam) [inline README.md:789]:
```suggestion
> For a standalone ESXI host managed by vCenter Server, comment or remove `vsphere_cluster`.
```

File: README.md [hunk 6]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> ```hcl
+> ...
+> vsphere_datacenter           = "sfo-w01-dc01"
+> //vsphere_cluster              = "sfo-w01-cl01"
+> vsphere_host                 = "sfo-w01-esx01"
+> vsphere_folder               = "sfo-w01-fd-templates"
+> ...
+> ```
+>
+> **Example** (Standalone Hosts):
```
Comment (tenthirtyam) [inline README.md:800]:
```suggestion
> **Example** (Standalone ESXi Host):
```

File: README.md [hunk 7]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+>
+> For standalone hosts, comment or remove `vsphere_datacenter`, `vsphere_cluster` and `vsphere_folder`.
```
Comment (tenthirtyam) [inline README.md:802]:
```suggestion
> For a standalone, unmanaged ESXi host, comment or remove `vsphere_datacenter`, `vsphere_cluster`, and `vsphere_folder`.
```

File: set-envvars.sh [hunk 8]
```diff
@@ -54,6 +54,7 @@ esac
 echo -e '\n> Set the vSphere settings.'
 read -r -p "Enter the vSphere datacenter name: " vsphere_datacenter
 read -r -p "Enter the vSphere cluster name: " vsphere_cluster
+read -r -p "Enter the vSphere host name: " vsphere_host
```
Comment (tenthirtyam) [inline set-envvars.sh:57]:
```suggestion
read -r -p "Enter the ESXi host FQDN or IP: " vsphere_host
```

File: README.md [hunk 9]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> ```
+>
+> **Example** (vCenter Managed Standalone Hosts):
+>
+> For vCenter managed standalone hosts, comment or remove `vsphere_cluster`.
+>
+> ```hcl
+> ...
+> vsphere_datacenter           = "sfo-w01-dc01"
+> //vsphere_cluster              = "sfo-w01-cl01"
```
Comment (tenthirtyam) [inline README.md:794]:
```suggestion
> //vsphere_cluster            = "sfo-w01-cl01"
```

File: README.md [hunk 10]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> ...
+> ```
+>
+> **Example** (Standalone Hosts):
+>
+> For standalone hosts, comment or remove `vsphere_datacenter`, `vsphere_cluster` and `vsphere_folder`.
+>
+> ```hcl
+> ...
+> //vsphere_datacenter           = "sfo-w01-dc01"
```
Comment (tenthirtyam) [inline README.md:806]:
```suggestion
> //vsphere_datacenter         = "sfo-w01-dc01"
```

File: README.md [hunk 11]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> //vsphere_cluster              = "sfo-w01-cl01"
```
Comment (tenthirtyam) [inline README.md:807]:
```suggestion
> //vsphere_cluster            = "sfo-w01-cl01"
```

File: README.md [hunk 12]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
...
+> vsphere_host                 = "sfo-w01-esx01"
+> //vsphere_folder               = "sfo-w01-fd-templates"
```
Comment (tenthirtyam) [inline README.md:809]:
```suggestion
> //vsphere_folder             = "sfo-w01-fd-templates"
```

File: README.md [hunk 13]
```diff
@@ -769,6 +769,47 @@ vsphere_network              = "sfo-w01-seg-dhcp"
 vsphere_folder               = "sfo-w01-fd-templates"
 ```
 
+> **Note**
```
Comment (tenthirtyam) [inline README.md:772]: In the example above, you should add 

```hcl
vsphere_endpoint             = "sfo-w01-vc01.sfo.rainpole.io"
vsphere_username             = "svc-packer-vsphere@rainpole.io"
vsphere_password             = "<plaintext_password>"
vsphere_insecure_connection  = true
vsphere_datacenter           = "sfo-w01-dc01"
vsphere_cluster              = "sfo-w01-cl01"
//vsphere_host               = "sfo-w01-esx01"
vsphere_datastore            = "sfo-w01-cl01-ds-vsan01"
vsphere_network              = "sfo-w01-seg-dhcp"
vsphere_folder               = "sfo-w01-fd-templates"
 ```

**Comment by tintinhamans** [comment]
Updated the wording to be as you suggested.

**Comment by tenthirtyam** [comment]
> Updated the wording to be as you suggested.

I'll test this next week in my labs.

**Comment by tenthirtyam** [comment]
I've rebased this pull requests and will review it soon.

**Comment by tintinhamans** [comment]
Fixed missing newline in `variables.pkr.hcl` files, added OSes that were added after initial pull request.

**Comment by tenthirtyam** [comment]
> Fixed missing newline in `variables.pkr.hcl` files, added OSes that were added after initial pull request.

Thank you! Should get back to this one next week (finally) due to other projects.

**Review by tenthirtyam — APPROVED** [review]
LGTM! I'll update the `CHANGELOG.md` and merge soon.

**Comment by tintinhamans** [comment]
While doing some more testing, discovered I forgot to add the folder defaults for the new OSes. That should be fixed now.

**Pull request merged**